warehouse/manage/forms.py

```diff
@@ -216,34 +221,41 @@ def validate_description(self, field):
         ):
             raise wtforms.validators.ValidationError("API token name already in use")
 
-    def validate_token_scope(self, field):
-        scope = field.data
+    def validate_token_scopes(self, field):
+        scopes = field.data
+        if not scopes:
+            raise wtforms.ValidationError(f"Specify the token scope")
 
-        try:
-            _, scope_kind = scope.split(":", 1)
-        except ValueError:
-            raise wtforms.ValidationError(f"Unknown token scope: {scope}")
+        self.validated_scope = {"projects": []}
 
-        if scope_kind == "unspecified":
-            raise wtforms.ValidationError(f"Specify the token scope")
+        for scope in scopes:
+            try:
+                _, scope_kind = scope.split(":", 1)
+            except ValueError:
+                raise wtforms.ValidationError(f"Unknown token scope: {scope}")
 
-        if scope_kind == "user":
-            self.validated_scope = scope_kind
-            return
+            if scope_kind == "unspecified":
+                raise wtforms.ValidationError(f"Specify the token scope")
 
-        try:
-            scope_kind, scope_value = scope_kind.split(":", 1)
-        except ValueError:
-            raise wtforms.ValidationError(f"Unknown token scope: {scope}")
-
-        if scope_kind != "project":
-            raise wtforms.ValidationError(f"Unknown token scope: {scope}")
-        if scope_value not in self.project_names:
-            raise wtforms.ValidationError(
-                f"Unknown or invalid project name: {scope_value}"
-            )
+            if scope_kind == "user":
+                if len(scopes) != 1:
+                    raise wtforms.ValidationError(f"Do not mix user and project scopes")
```

Comment:
Nit: Maybe drop the imperative?

```python
raise wtforms.ValidationError("Mixed user and project scopes")
```

Reply:
In addition to this, we could probably also add some JS to prevent the user from selecting both project scopes and the user scope on the client side.

Reply:
That sounds like a good next step but I can barely read JS so would need some guidance on how to make that happen :P